src/Components/Feature/ToDoListRq.tsx

```diff
@@ -0,0 +1,138 @@
+    const [pageNumber, setPageNumber] = useState<any>(0)
+    const [originalTodos, setOriginalFilteredTodos] = useState<IToDoResp[]>([]);
+    const [filteredTodos, setFilteredTodos] = useState<IToDoResp[]>([]);
+    const [searchTerm, setSearchTerm] = useState<string>('');
+    const [sortBy, setSortBy] = useState<string>('');
+    const [statusFilter, setStatusFilter] = useState<string>('');
+
+    const onSuccess =  (data:IToDoResp[]) =>{
+        console.log(data)
+        setFilteredTodos(data);
+        setOriginalFilteredTodos(data)
+    }
+
+    const { isLoading, isError,error } = useQuery(['todos',pageNumber], ()=>fetchTodos(pageNumber),{
+        onSuccess,
+        refetchOnWindowFocus:false,
+        keepPreviousData:true
+    });
+
+
+    const markCompleted = (id: number) => {
+        if (filteredTodos) {
+          const updatedTodos:IToDoResp[] | null = filteredTodos.map(todo =>
+            todo.id === id ? { ...todo, completed: !todo.completed } : todo
+          );
+          setFilteredTodos(updatedTodos);
+        }
+      };
+    
+      const deleteTodo = (id: number) => {
+        if (filteredTodos) {
+          const updatedTodos = filteredTodos.filter(todo => todo.id !== id);
+          setFilteredTodos(updatedTodos);
+        }
+      };
+    
+        // Handle search input
+        const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
+          setSearchTerm(e.target.value);
+          filterTodos(e.target.value, sortBy, statusFilter);
+      };
+    
+      // Handle sort dropdown
+      const handleSort = (e: ChangeEvent<HTMLSelectElement>) => {
+          setSortBy(e.target.value);
+          filterTodos(searchTerm, e.target.value, statusFilter);
+      };
+    
+      // Handle status dropdown
+      const handleStatus = (e: ChangeEvent<HTMLSelectElement>) => {
+          setStatusFilter(e.target.value);
+          filterTodos(searchTerm, sortBy, e.target.value);
+      };
+    
+      // Filter todos based on search term, sort, and status
+      const filterTodos = (search: string, sort: string, status: string) => {
+          let filtered = filteredTodos ?? []
+    
+          if(!search){
+            setFilteredTodos(originalTodos);
+            filtered = originalTodos
+          }
+          else{
+            filtered = filtered.filter(todo =>
+              todo.title.toLowerCase().includes(search.toLowerCase())
+            );
+          }
+    
+          if (sort === 'asc') {
+              filtered.sort((a, b) => a.id - b.id);
+          } else if (sort === 'desc') {
+              filtered.sort((a, b) => b.id - a.id);
+          }
+    
+          if (status) {
+              filtered = filtered.filter(todo => todo.completed === (status === 'completed'));
+          }
+    
+          setFilteredTodos(filtered);
+      };
+
+      if (isLoading) return <div>Loading...</div>;
+      if (isError) return <div>Error fetching data</div>;
+
+      return (
+        <div className='todo-list'>
```

Comment:
move this JSX into the component. Here we are making API call consider this is an container